tests/test_pipeline/test_hierarchical_pipeline.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from etna.datasets import TSDataset
+from etna.metrics import MAE
+from etna.models import MovingAverageModel
+from etna.pipeline.hierarchical_pipeline import HierarchicalPipeline
+from etna.reconciliation import BottomUpReconciliator
+from etna.reconciliation import TopDownReconciliator
+
+
+@pytest.fixture
+def market_level_constant_hierarchical_ts(hierarchical_structure):
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02", "2000-01-03"] * 2,
+            "segment": ["X"] * 3 + ["Y"] * 3,
+            "target": [2, 2, 2] + [2, 2, 2],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    ts = TSDataset(df=df, freq="D", hierarchical_structure=hierarchical_structure)
+    return ts
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="a", source_level="b", period=1, method="AHP"),
+        BottomUpReconciliator(target_level="a", source_level="b"),
+    ),
+)
+def test_init_pass(reconciliator):
+    model = MovingAverageModel(window=1)
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=1)
+    assert isinstance(pipeline.reconciliator, type(reconciliator))
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="market", source_level="total", period=1, method="AHP"),
+        BottomUpReconciliator(target_level="total", source_level="market"),
+    ),
+)
+def test_fit_mapping_matrix(market_level_simple_hierarchical_ts, reconciliator):
+    model = MovingAverageModel(window=1)
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=1)
+    pipeline.fit(market_level_simple_hierarchical_ts)
+    assert pipeline.reconciliator.mapping_matrix is not None
```

Comment:
Here you can use [mocks](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called) to check that method `fit` of reconciliator was called